tests/visualisations/test_trackmap.py

```diff
@@ -19,16 +28,109 @@ def prefilled_trackmap(example_plottingdataframe):
 
 
 def test_TrackMap(emptytrackmap, prefilled_trackmap, example_plottingdataframe):
-    objects_in_empty = len(emptytrackmap.__dict__['_children'])
-    objects_in_nonempty = len(prefilled_trackmap.__dict__['_children'])
+    objects_in_empty = len(emptytrackmap._children)
+    objects_in_nonempty = len(prefilled_trackmap._children)
     assert objects_in_empty == 1, 'Should contain only the aerial photo'
-    assert objects_in_nonempty == 1 + len(example_plottingdataframe), 'Should contain the aerial photo and some markers'
+    assert objects_in_nonempty == 1 + len(
+        example_plottingdataframe.data), 'Should contain the aerial photo and some markers'
+
+
+@pytest.fixture
+def points_dataframe():
+    return pd.DataFrame(data={'lat': [52.45, 52.5, 52.55, 52.6], 'lon': [5.15, 5.2, 5.3, 5.4],
+                              'name': ['A', 'B', "C", 'D'],
+                              'name2': ['AA', 'BB', "CC", 'DD'],
+                              'value': [1, 2, 3, 4],
+                              "marker": ['train', "eye", 'train', "eye"]
+                              })
+
+
+@pytest.fixture
+def lines_geodataframe():
+    line1 = LineString([Point(4.9, 52.1), Point(4.95, 51.1)]).wkt
+    line2 = LineString([Point(5.0, 52.1), Point(4.95, 51.1)]).wkt
+    line3 = LineString([Point(4.8, 52.1), Point(4.95, 51.1)]).wkt
+
+    return (
+        pd.DataFrame(data={'name': ['Q', 'R', 'S'],
+                           'name2': ['QQ', 'RR', 'SS'],
+                           'geometry': [line1, line2, line3]})
+        .assign(geometry=lambda d: d.geometry.apply(wkt.loads))
+        .pipe(gpd.GeoDataFrame, geometry='geometry', crs='EPSG:4326')
+        .assign(geometry=lambda d: d.geometry.to_crs('EPSG:4326'))
+    )
+
+
+@pytest.fixture
+def areas_geodataframe():
+    area1 = Polygon([Point(5.0, 52.0), Point(5.1, 52.0), Point(5.1, 52.1), Point(5.0, 52.1)]).wkt
+    area2 = Polygon([Point(5.5, 52.0), Point(5.6, 52.0), Point(5.6, 52.1), Point(5.5, 52.1)]).wkt
+
+    return (
+        pd.DataFrame(data={'name': ['M', 'N'], 'name2': ['MM', 'NN'], 'geometry': [area1, area2]})
+        .assign(geometry=lambda d: d.geometry.apply(wkt.loads))
+        .pipe(gpd.GeoDataFrame, geometry='geometry', crs='EPSG:4326')
+        .assign(geometry=lambda d: d.geometry.to_crs('EPSG:4326'))
+    )
+
+
+def test_add_to_trackmap(tmp_path, points_dataframe, lines_geodataframe, areas_geodataframe):
+    m = TrackMap()
+
+    PlottingPoints(points_dataframe, popup=['name2']).add_to(m)
+    PlottingLineStrings(lines_geodataframe, popup=['name2']).add_to(m)
+    PlottingAreas(areas_geodataframe, popup=['name2']).add_to(m)
+    total_objects = len(points_dataframe) + len(lines_geodataframe) + len(areas_geodataframe)
+
+    assert len(m._children) == total_objects, "Invalid number of items added"
```

Comment:
Should this not be: total_objects +1, where the +1 is the aerial photograph? You are adding 4 points, 3 lines and 2 areas.

Reply:
This is not the case, the number of Children is not equal to the number of points + lines + areas. However, for lines and areas a Choropleth and Geojson child are added instead of 3x a line

Reply:
Nice catch. As discussed I will update this.